cv.py

```diff
@@ -62,89 +87,60 @@ def run_cv(self):
         mask = cv2.inRange(hsv, low_green, high_green)
         _, contours,_ = cv2.findContours(mask, cv2.RETR_TREE, cv2.CHAIN_APPROX_SIMPLE)
 
-        contours.sort(key=cv2.contourArea, reverse=True)
-        maxContours = []
-        rotatedBoxes = []
+        #sorts contours by area
+        contours.sort(key = cv2.contourArea, reverse = True)
+
+        rotated_boxes = []
+        rotated_rect1 = None
+        rotated_rect2 = None
+        rotated_rect3 = None
+        self.pinX = 0
+        self.pinY = 0
 
         for c in contours[:3]:
-            x, y, w, h = cv2.boundingRect(c)
             area = cv2.contourArea(c)
-            maxContours.append(Rectangle(x, y, w, h, area))
-            if(area > 500):
-                rect = cv2.minAreaRect(c)
-                _,_, rot_angle = rect
-                box = cv2.boxPoints(rect)
-                box = np.int0(box)
-                rotatedBoxes.append(RotatedRectangle(box, area, rot_angle))
-
-        if(len(rotatedBoxes) > 1):
-            rotated_rect1 = rotatedBoxes[0]
-            rotated_rect2 = rotatedBoxes[1]
-
-            cv2.drawContours(frame, [rotated_rect1.box], 0, (0, 0, 255), 2)
-            # cv2.putText(frame, "Rect1: " + str(rotated_rect1.rot_angle), (rotated_rect1.box[0][0], rotated_rect1.box[0][1]), cv2.FONT_HERSHEY_DUPLEX, 2, 255)
+            rect = cv2.minAreaRect(c)
+            _,_, rot_angle = rect
+            box = cv2.boxPoints(rect)
+            box = np.int0(box)
+            if area > 100:
+                rotated_boxes.append(RotatedRectangle(box, area, rot_angle))
 
-            cv2.drawContours(frame, [rotated_rect2.box], 0, (0, 0, 255), 2)
-            # cv2.putText(frame, "Rect2: " + str(rotated_rect2.rot_angle), (rotated_rect2.box[0][0], rotated_rect2.box[0][1]), cv2.FONT_HERSHEY_DUPLEX, 2, 255)
+        if(len(rotated_boxes) > 1):
+            rotated_rect1 = rotated_boxes[0]
+            rotated_rect2 = rotated_boxes[1]
```

Comment:
I'm not sure why you're giving distinct names to the individual elements of arrays. It's very easy to mix up copies with the original, and I'm not sure it's any more readable.